Install fails with node 10

<details>
<summary>Error log</summary>

```
warning Error running install script for optional dependency: "/Users/jetbrains/Perforce/Filipp.Riabchun_unit-951_3393/react-ui/node_modules/png-img: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /Users/jetbrains/Perforce/Filipp.Riabchun_unit-951_3393/react-ui/node_modules/png-img
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.6.2
gyp info using node@10.0.0 | darwin | x64
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp info spawn /usr/bin/python
gyp info spawn args [ '/Users/jetbrains/.nvm/versions/node/v10.0.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/jetbrains/Perforce/Filipp.Riabchun_unit-951_3393/react-ui/node_modules/png-img/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/jetbrains/.nvm/versions/node/v10.0.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/jetbrains/.node-gyp/10.0.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/jetbrains/.node-gyp/10.0.0',
gyp info spawn args   '-Dnode_gyp_dir=/Users/jetbrains/.nvm/versions/node/v10.0.0/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/jetbrains/.node-gyp/10.0.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/jetbrains/Perforce/Filipp.Riabchun_unit-951_3393/react-ui/node_modules/png-img',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CC(target) Release/obj.target/libpng/third_party/libpng/png.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngerror.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngget.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngmem.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngpread.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngread.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngrio.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngrtran.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngrutil.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngset.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngtest.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngtrans.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngwio.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngwrite.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngwtran.o
  CC(target) Release/obj.target/libpng/third_party/libpng/pngwutil.o
  LIBTOOL-STATIC Release/png.a
  CC(target) Release/obj.target/zlib/third_party/zlib/adler32.o
  CC(target) Release/obj.target/zlib/third_party/zlib/compress.o
  CC(target) Release/obj.target/zlib/third_party/zlib/crc32.o
  CC(target) Release/obj.target/zlib/third_party/zlib/deflate.o
  CC(target) Release/obj.target/zlib/third_party/zlib/gzclose.o
  CC(target) Release/obj.target/zlib/third_party/zlib/gzlib.o
  CC(target) Release/obj.target/zlib/third_party/zlib/gzread.o
  CC(target) Release/obj.target/zlib/third_party/zlib/gzwrite.o
  CC(target) Release/obj.target/zlib/third_party/zlib/infback.o
  CC(target) Release/obj.target/zlib/third_party/zlib/inffast.o
  CC(target) Release/obj.target/zlib/third_party/zlib/inflate.o
../third_party/zlib/inflate.c:1507:61: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
    if (strm == Z_NULL || strm->state == Z_NULL) return -1L << 16;
                                                        ~~~ ^
1 warning generated.
  CC(target) Release/obj.target/zlib/third_party/zlib/inftrees.o
  CC(target) Release/obj.target/zlib/third_party/zlib/trees.o
  CC(target) Release/obj.target/zlib/third_party/zlib/uncompr.o
  CC(target) Release/obj.target/zlib/third_party/zlib/zutil.o
  LIBTOOL-STATIC Release/zlib.a
  CXX(target) Release/obj.target/png_img/src/init.o
In file included from ../src/init.cc:1:
../node_modules/nan/nan.h:839:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:171:1: note: 'MakeCallback' has been explicitly marked deprecated here
NODE_DEPRECATED(\"Use MakeCallback(..., async_context)\",
^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../src/init.cc:1:
../node_modules/nan/nan.h:854:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:164:1: note: 'MakeCallback' has been explicitly marked deprecated here
NODE_DEPRECATED(\"Use MakeCallback(..., async_context)\",
^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../src/init.cc:1:
../node_modules/nan/nan.h:869:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:157:1: note: 'MakeCallback' has been explicitly marked deprecated here
NODE_DEPRECATED(\"Use MakeCallback(..., async_context)\",
^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../src/init.cc:1:
../node_modules/nan/nan.h:1648:31: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
    return scope.Escape(node::MakeCallback(
                              ^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:171:1: note: 'MakeCallback' has been explicitly marked deprecated here
NODE_DEPRECATED(\"Use MakeCallback(..., async_context)\",
^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
4 warnings generated.
  CXX(target) Release/obj.target/png_img/src/PngImgAdapter.o
In file included from ../src/PngImgAdapter.cc:1:
In file included from ../src/./PngImgAdapter.h:4:
../node_modules/nan/nan.h:839:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:171:1: note: 'MakeCallback' has been explicitly marked deprecated here
NODE_DEPRECATED(\"Use MakeCallback(..., async_context)\",
^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../src/PngImgAdapter.cc:1:
In file included from ../src/./PngImgAdapter.h:4:
../node_modules/nan/nan.h:854:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:164:1: note: 'MakeCallback' has been explicitly marked deprecated here
NODE_DEPRECATED(\"Use MakeCallback(..., async_context)\",
^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../src/PngImgAdapter.cc:1:
In file included from ../src/./PngImgAdapter.h:4:
../node_modules/nan/nan.h:869:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:157:1: note: 'MakeCallback' has been explicitly marked deprecated here
NODE_DEPRECATED(\"Use MakeCallback(..., async_context)\",
^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../src/PngImgAdapter.cc:1:
In file included from ../src/./PngImgAdapter.h:4:
../node_modules/nan/nan.h:1648:31: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
    return scope.Escape(node::MakeCallback(
                              ^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:171:1: note: 'MakeCallback' has been explicitly marked deprecated here
NODE_DEPRECATED(\"Use MakeCallback(..., async_context)\",
^
/Users/jetbrains/.node-gyp/10.0.0/include/node/node.h:88:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
../src/PngImgAdapter.cc:52:65: error: no matching member function for call to 'NewInstance'
    info.GetReturnValue().Set(constructorHandle->GetFunction()->NewInstance(1, argv));
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
/Users/jetbrains/.node-gyp/10.0.0/include/node/v8.h:3851:44: note: candidate function not viable: requires single argument 'context', but 2 arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                           ^
/Users/jetbrains/.node-gyp/10.0.0/include/node/v8.h:3848:44: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                           ^
../src/PngImgAdapter.cc:102:27: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
        String::Utf8Value val(obj->Get(Nan::New<String>(key.c_str()).ToLocalChecked()));
                          ^
/Users/jetbrains/.node-gyp/10.0.0/include/node/v8.h:2821:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED(\"Use Isolate version\",
    ^
/Users/jetbrains/.node-gyp/10.0.0/include/node/v8config.h:321:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/PngImgAdapter.cc:188:53: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
    AsyncQueueWorker(new SaveWorker(callback, img, *v8::String::Utf8Value(file), info.This()));
                                                    ^
/Users/jetbrains/.node-gyp/10.0.0/include/node/v8.h:2821:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED(\"Use Isolate version\",
    ^
/Users/jetbrains/.node-gyp/10.0.0/include/node/v8config.h:321:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
6 warnings and 1 error generated.
make: *** [Release/obj.target/png_img/src/PngImgAdapter.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/jetbrains/.nvm/versions/node/v10.0.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:225:12)
gyp ERR! System Darwin 17.5.0
gyp ERR! command \"/Users/jetbrains/.nvm/versions/node/v10.0.0/bin/node\" \"/Users/jetbrains/.nvm/versions/node/v10.0.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js\" \"rebuild\"
gyp ERR! cwd /Users/jetbrains/Perforce/Filipp.Riabchun_unit-951_3393/react-ui/node_modules/png-img
```

</details>

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,6 +4,9 @@ sudo: false
 node_js:
   - '4'
   - '5'
+  - '6'
+  - '8'
+  - '10'
 
 env:
   - CXX=g++-4.8

--- a/src/PngImgAdapter.cc
+++ b/src/PngImgAdapter.cc
@@ -101,7 +101,7 @@ NAN_METHOD(PngImgAdapter::Get) {
 ///
 Pxl RGBObjToPxl(const Local<Object>& obj) {
     auto getIntVal_ = [&obj](const string& key) {
-        String::Utf8Value val(obj->Get(Nan::New<String>(key.c_str()).ToLocalChecked()));
+        Nan::Utf8String val(obj->Get(Nan::New<String>(key.c_str()).ToLocalChecked()));
         return stoi(*val);
     };
 
@@ -187,7 +187,7 @@ NAN_METHOD(PngImgAdapter::Write) {
     Local<String> file = info[0].As<String>();
     Callback* callback = new Callback(info[1].As<Function>());
 
-    AsyncQueueWorker(new SaveWorker(callback, img, *v8::String::Utf8Value(file), info.This()));
+    AsyncQueueWorker(new SaveWorker(callback, img, *Nan::Utf8String(file), info.This()));
     info.GetReturnValue().SetUndefined();
 }
 

--- a/src/PngImgAdapter.cc
+++ b/src/PngImgAdapter.cc
@@ -49,7 +49,9 @@ NAN_METHOD(PngImgAdapter::NewInstance) {
     Local<FunctionTemplate> constructorHandle = Nan::New(pngImgAdapterConstructor);
     Local<Object> imgBuffer = info[0].As<Object>();
     Handle<Value> argv[] = { imgBuffer };
-    info.GetReturnValue().Set(constructorHandle->GetFunction()->NewInstance(1, argv));
+
+    Local<Object> instance = Nan::NewInstance(constructorHandle->GetFunction(), 1, argv).ToLocalChecked();
+    info.GetReturnValue().Set(instance);
 }
 
 ///

